gdal/frmts/sentinel2/sentinel2dataset.cpp

```diff
@@ -179,7 +186,8 @@ class SENTINEL2Dataset : public VRTDataset
                 bool bAlpha,
                 const std::vector<CPLString>& aosBands,
                 int nSaturatedVal,
-                int nNodataVal);
+                int nNodataVal,
+                const CPLString pszProductURI);
```

Comment:
Naming convention: should be osProductURI (psz is for 'p'ointer to a 's'tring 'z'ero terminated,  os is for 'o'bject 's'tring)
And calling convention: should be const CPLString& to pass by reference

Reply:
See https://trac.osgeo.org/gdal/wiki/rfc8_devguide